crates/bytes-util/src/zero_copy/serde.rs

```diff
@@ -0,0 +1,370 @@
+    {
+        writer.write_all(self.as_bytes())
+    }
+}
+
+impl<'a> Deserialize<'a> for StringCow<'a> {
+    fn deserialize<R>(mut reader: R) -> io::Result<Self>
+    where
+        R: ZeroCopyReader<'a>,
+    {
+        let bytes = reader.try_read_to_end()?;
+        Self::try_from(bytes).map_err(|_| io::Error::new(io::ErrorKind::InvalidData, "Invalid UTF-8"))
+    }
+}
+
+impl Serialize for StringCow<'_> {
+    fn serialize<W>(&self, mut writer: W) -> io::Result<()>
+    where
+        W: std::io::Write,
+    {
+        writer.write_all(self.as_str().as_bytes())
+    }
+}
+
+/// A 24-bit signed integer in big-endian byte order.
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+pub struct I24Be(pub i32);
+
+impl<'a> Deserialize<'a> for I24Be {
+    fn deserialize<R: ZeroCopyReader<'a>>(mut reader: R) -> io::Result<Self> {
+        reader.as_std().read_i24::<byteorder::BigEndian>().map(I24Be)
+    }
+}
+
+impl Serialize for I24Be {
+    fn serialize<W>(&self, mut writer: W) -> io::Result<()>
+    where
+        W: std::io::Write,
+    {
+        writer.write_all(&self.0.to_be_bytes()[1..])
+    }
+}
+
+impl From<I24Be> for i32 {
+    fn from(value: I24Be) -> Self {
+        value.0
+    }
+}
+
+impl Deref for I24Be {
+    type Target = i32;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// A 48-bit signed integer in big-endian byte order.
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+pub struct I48Be(pub i64);
+
+impl<'a> Deserialize<'a> for I48Be {
+    fn deserialize<R: ZeroCopyReader<'a>>(mut reader: R) -> io::Result<Self> {
+        reader.as_std().read_i48::<byteorder::BigEndian>().map(I48Be)
+    }
+}
+
+impl Serialize for I48Be {
+    fn serialize<W>(&self, mut writer: W) -> io::Result<()>
+    where
+        W: std::io::Write,
+    {
+        writer.write_all(&self.0.to_be_bytes()[2..])
+    }
+}
+
+impl From<I48Be> for i64 {
+    fn from(value: I48Be) -> Self {
+        value.0
+    }
+}
+
+impl Deref for I48Be {
+    type Target = i64;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// A 24-bit unsigned integer in big-endian byte order.
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+pub struct U24Be(pub u32);
+
+impl<'a> Deserialize<'a> for U24Be {
+    fn deserialize<R: ZeroCopyReader<'a>>(mut reader: R) -> io::Result<Self> {
+        reader.as_std().read_u24::<byteorder::BigEndian>().map(U24Be)
+    }
+}
+
+impl Serialize for U24Be {
+    fn serialize<W>(&self, mut writer: W) -> io::Result<()>
+    where
+        W: std::io::Write,
+    {
+        writer.write_all(&self.0.to_be_bytes()[1..])
+    }
+}
+
+impl From<U24Be> for u32 {
+    fn from(value: U24Be) -> Self {
+        value.0
+    }
+}
+
+impl Deref for U24Be {
+    type Target = u32;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// A 48-bit unsigned integer in big-endian byte order.
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+pub struct U48Be(pub u64);
```

Comment:
Like the u24, no seemingly premade u48 number type. https://github.com/Chubercik/i48/issues/5